api/src/server/gql/mutation.rs

```diff
@@ -239,14 +240,19 @@ impl MutationFields for Mutation {
         };
         modified_user_program.validate()?;
 
-        // Update the row, returning the new value. If the row doesn't exist,
-        // this will return None.
-        let result: Result<Option<models::UserProgram>, _> =
-            diesel::update(user_programs::table.find(modified_user_program.id))
-                .set(modified_user_program)
-                .returning(user_programs::table::all_columns())
-                .get_result(conn)
-                .optional();
+        // Update the row, returning the new value. If the row doesn't
+        // exist, this will return None.
+        let result: Result<Option<models::UserProgram>, _> = diesel::update(
+            user_programs::table
+                .find(modified_user_program.id)
+                // Extra filter to make sure the requesting user is the owner of
+                // the row.
+                .filter(user_programs::columns::user_id.eq(user.id)),
+        )
```

Comment:
Should we throw an error saying you dont have access to this program instead of doing nothing?

Reply:
I think in general, if a user doesn't have read perms on an object, then any time they try to do something to it we should pretend like it doesn't exist, to prevent leaking any data. In this case it wouldn't be a big deal cause we would just be leaking that a certain guid exists in the DB, but i think it's still the best practice.

Reply:
![image](https://user-images.githubusercontent.com/5672686/86512669-187f5000-bdd2-11ea-9df5-8207b341488a.png)